### DISCUSSION TITLE
Matchers in expectJsonSnapshot not working?

### DISCUSSION THREAD

**Discussion by captaincc423** [pactumjs/pactum#98] [Q&A] [answered]
I'm trying to get some matchers to work in expectJsonSnapshot and I'm running into a few problems:

1. I'm trying to use a variable variable with a string in an "includes()", e.g. `"id": includes(response.body.id)` and it's not working. It always says it doesn't match.
2. I'm using a regex and it appears that no matter what I put in it, it always matches. For instance, `"createdAt": regex(/anything in here/),` always passes.

Am I misusing these matchers?

**Comment by ASaiAnudeep** [comment, marked as answer]
###### Welcome to PactumJS Discussions & thanks a lot for using this library.

1. I have tested `includes` and it is working as expected. It you could provide more info on this, it would help us to investigate further.
2. There was a typo in our docs on the usage of `regex`. The function accepts two arguments - value & regex. Will update the docs and the function to accept a single argument.

```js
it('first-post', async () => {
  await pactum.spec().name('first-post')
    .get('https://jsonplaceholder.typicode.com/posts/1')
    .expectJsonSnapshot({
      title: includes("provident"),
      id: regex(1, /\d+/)
    });
});
```

**Reply by captaincc423** [reply]
1. For some reason if I type the string in directly like this: `let name = "My Expected name";` it's working fine, but I'm trying to use the value from a previous response like this `let id = response.body.id;` and it won't work. I can print it out and see it the the value is correct, but the matcher always fails. It says:

`AssertionError [ERR_ASSERTION]: Json doesn't match with "id_from_response" at "$.data[0].id" but found "id_from_snapshot"`

2. Thank you. I'll wait on that update!

**Reply by ASaiAnudeep** [reply]
There are two steps in validating the json snapshots.

1. Comparing the actual response with the expected snapshot with the given matching rules. Here `includes('dynamic content')` will be successfully matched.
2. Comparing the expected snapshot with the actual response with the given matching rules. This is vice-versa of the first step and sounds confusing. Here `includes('dynamic content')` will fail due to the saved snapshot in the file system will contain old content.

When `expectJsonSnapshot` is introduced, the matcher `includes` is not part of this library. Seems both of them are not compatible with each other in some corner cases.

Created an issue #100 to fix it.

As a work around, please use `like` or `regex` matcher if it fits your requirement.
